### ISSUE TITLE
pickadate fields with type: Number are not formatting MM/YYYY in autoform 'Update' forms

### ISSUE THREAD

**Issue by 0o-de-lally** [djhi/meteor-autoform-materialize#92]
I'm using pickadate form fields to collect data. When the user inputs a new clean field, the date appears to the user as MM/YYYY. Good.

However, when the user returns to edit that same field, it appears as unix epoch milliseconds 14000000000 . It seems not to render.

```
  startDate : {
    type: Number,
    autoform: {
    type:"pickadate"
    }
 },
```

Am I doing something wrong? Is there a work around? Or is this not implemented?

(I must use milliseconds for this dataset. It's part of a large system that uses milliseconds across components)

**Comment by 0o-de-lally** [comment]
Correction: I noticed that the correct formatting MM/YYYY happens only when the page is refreshed. (Perhaps only when the page is rendered). If a reactive variable, or session is used to update a form (as I am doing) then the field will not format correctly.
So I'm guessing this formatting is happening onRender, instead of with a Tracker.autorun ?

**Comment by 0o-de-lally** [comment]
@serkandurusoy I see you worked on 'pickadate' type recently. Any ideas? Thanks in advance :)

**Comment by serkandurusoy** [comment]
I've forked the repo and applied https://github.com/djhi/meteor-autoform-materialize/pull/85 and comment out `else {picker.set('clear');}` so that picker persists across invalid submissions

 I also use a `pickadateOptions` object on `autoform` to pass in a `format` parameter

And finally I work with `Date` types not milliseconds.

**Comment by 0o-de-lally** [comment]
I think my problem applies to all types, not just milliseconds. I'm wondering if the format parameter is being ignored when I modify reactively the autoform 'update' doc. That is... it's only being applied on render. Does this make sense?

**Comment by serkandurusoy** [comment]
Did you try this:

``` javascript
startDate : {
    type: Number,
    autoform: {
      type:"pickadate",
      pickadateOptions: {
        format: "MM/YYYY" // I don't know if this is a valid format, though, I use "dd.mm.yyyy"
      }
    }
 }
```

**Comment by 0o-de-lally** [comment]
@serkandurusoy  Sorry no change with that code.
To be clear: I do get the correct format when I refresh, or change routes. It works perfectly.
However if I change the form doc reactively with a Session.get ... that when date fields lose their format. That's why I wonder if the field formatter should trigger on autorun, and not only on render.

Some context: In my UI,  on the left: I have a list, on the right:  a form. The list has form options. When they chose one of the list options, that's when form defaults change.

Thanks for your help!

**Comment by serkandurusoy** [comment]
hmm, well, you can set up an autorun that depends on your session and then calls `$('/*classname for your datepickers*/').pickadate()`

**Comment by 0o-de-lally** [comment]
@serkandurusoy Ok, but I think other people are likely having the same issue, even if it's not milliseconds. I'm thinking the issue is on line 70 here https://github.com/djhi/meteor-autoform-materialize/blob/master/inputTypes/pickadate/pickadate.js 

The autorun should probably include the other variables set on render... but I'm not familiar enough with the code.

**Comment by serkandurusoy** [comment]
Hmm line 89 may be the culprit since it runs only once because it is already satisfied, but it does satisfy general usecases.

For your specific case, though, I'd really suggest setting up an external autorun just like I suggested in my previous case.

**Comment by 0o-de-lally** [comment]
@serkandurusoy  Thanks again for this. I'm trying without much luck.
Here's a repo that can reproduce this. https://github.com/keyscores/autoform-materialize-datepicker-issue

I think other people will have this issue, it's not just Number types. Thanks in advance for taking a look at it!

**Comment by 0o-de-lally** [comment]
@serkandurusoy @Chun-Yang 

On line 89 I made a change to make milliseconds work.

   if (data.value instanceof Date || !isNaN( data.value )  ) {

However this will still break for other types you are converting from. 'String' 'Array' are other possible types documented. https://github.com/djhi/meteor-autoform-materialize#automatic-type-conversions

**Comment by djhi** [comment]
Do you guys want to be added as contributors on this projet and be able to publish new versions ?

**Comment by 0o-de-lally** [comment]
@djhi Thanks for the offer. I'm not sure I can be more helpful besides these bug reports. I'm happy to submit a pull request but I think it's incomplete. Any ideas how to improve?

**Comment by djhi** [comment]
Honestly I don't know, I don't have time anymore for meteor projects. We don't use it at my company, at least for now. I didn't have much knowledge about js and meteor at the time I wrote this so don't be shy and try !

**Comment by 0o-de-lally** [comment]
I'll send a pull request if I can solve it. Right now I have a memory leak, which I'm not sure is related to this.

**Comment by serkandurusoy** [comment]
Thanks for the offer, but I am past materializecss and will be using bootstrap for blaze/autoform projects and if I decide to use material ui, that'll probably be on react, going forward.

But I do appreciate all the work and contributions you have done. This package has been very valuable to me on a project I take pride in.

**Comment by djhi** [comment]
Future work on this package has been transferred [here](https://github.com/mozfet/meteor-autoform-materialize).
